hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/config/RateProperties.java

```diff
@@ -0,0 +1,98 @@
+package pl.allegro.tech.hermes.consumers.config;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import pl.allegro.tech.hermes.consumers.consumer.rate.calculator.RateCalculatorParameters;
+
+@ConfigurationProperties(prefix = "consumer.rate")
+public class RateProperties {
+
+    private int limiterSupervisorPeriod = 30;
+
+    private int limiterReportingThreadPoolSize = 30;
+
+    private boolean limiterReportingThreadMonitoringEnabled = false;
+
+    private int limiterHeartbeatModeDelay = 60;
+
+    private int limiterSlowModeDelay = 60;
+
+    private double convergenceFactor = 0.2;
```

Comment:
just asking. Why do you decided to use double instead of float?

Reply:
Thats because a double is a more accurate way of representing floating point numbers due to more digits of precision, and it is also the standard format in math processing units these days. As a result, defaulting to a double for constants will yield more accurate and consistent answers.